spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,239 @@
+    """
+    Stream IBL data as an extractor object.
+
+    Parameters
+    ----------
+    session : str
+        The session ID to extract recordings for.
+        In ONE, this is sometimes referred to as the 'eid'.
+        When doing a session lookup such as
+
+        >>> from one.api import ONE
+        >>> one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+        >>> sessions = one.alyx.rest('sessions', 'list', tag='2022_Q2_IBL_et_al_RepeatedSite')
+
+        each returned value in `sessions` refers to it as the 'id'.
+    stream_name : str
+        The name of the stream to load for the session.
+        These can be retrieved from calling `StreamingIblExtractor.get_stream_names(session="<your session ID>")`.
+    load_sync_channels : bool, default: false
+        Load or not the last channel (sync).
+        If not then the probe is loaded.
+    cache_folder : str, optional
+        The location to temporarily store chunks of data during streaming.
+        The default uses the folder designated by ONE.alyx._par.CACHE_DIR / "cache".
+    remove_cached : bool, default: true
+        Whether or not to remove data from the cache after it is read.
+        If you expect to reuse fetched data many times, and have the disk space available, it is recommended to set this to False.
+
+    Returns
+    -------
+    recording : StreamingIblExtractor
+        The recording extractor which allows access to the traces.
+    """
+
+    extractor_name = "StreamingIbl"
+    has_default_locations = True
+    installed = HAVE_BRAINBOX_ONE
+    mode = "folder"
+    installation_mesg = (
+        "To use the StreamingIblExtractor, install ONE-api and ibllib: \n\n pip install ONE-api\npip install ibllib\n"
+    )
+    name = "stream_ibl"
+
+    @classmethod
+    def get_stream_names(cls, session: str, cache_folder: Optional[Union[Path, str]] = None) -> List[str]:
+        """
+        Convenient retrieval of available stream names.
+
+        Parameters
+        ----------
+        session : str
+            The session ID to extract recordings for.
+            In ONE, this is sometimes referred to as the 'eid'.
+            When doing a session lookup such as
+
+            >>> from one.api import ONE
+            >>> one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+            >>> sessions = one.alyx.rest('sessions', 'list', tag='2022_Q2_IBL_et_al_RepeatedSite')
+
+            each returned value in `sessions` refers to it as the 'id'.
+
+        Returns
+        -------
+        stream_names : list of str
+            List of stream names as expected by the `stream_name` argument for the class initialization.
+        """
+        assert HAVE_BRAINBOX_ONE, cls.installation_mesg
+        one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True, cache_dir=cache_folder)
```

Comment:
One small change that I propagated after your last approval - I noticed it was storing session info in the default cache dir even when using a specified cache_folder